doc/pantry.md

```diff
@@ -0,0 +1,482 @@
+  git: git@github.com:yesodweb/wai
+  pantry-tree:
+    size: 687
+    sha256: 26377897f35ccd3890b4405d72523233717afb04d62f2d36031bf6b18dcef74f
+  commit: 2f8a8e1b771829f4a8a77c0111352ce45a14c30f
+- subdir: wai
+  cabal-file:
+    size: 1717
+    sha256: 7b46e7a8b121d668351fa8a684810afadf58c39276125098485203ef274fd056
+  name: wai
+  version: 3.0.2.3
+  git: git@github.com:yesodweb/wai
+  pantry-tree:
+    size: 10299
+    sha256: ce33fddab13592c847fbd7acd1859dfcbb9aeb6c212db3cee27c909fa3f3ae44
+  commit: 2f8a8e1b771829f4a8a77c0111352ce45a14c30f
+```
+
+### Archives (HTTP(S) or local filepath)
+
+You can use HTTP and HTTPS URLs and local filepaths referring to
+either tarballs or ZIP files.
+
+__NOTE__ Stack assumes that these files never change after downloading
+to avoid needing to make an HTTP request on each build. Use hashes to
+provide more security.
+
+```yaml
+extra-deps:
+- https://example.com/foo/bar/baz-0.0.2.tar.gz
+- archive: http://github.com/yesodweb/wai/archive/2f8a8e1b771829f4a8a77c0111352ce45a14c30f.zip
+  subdirs:
+  - wai
+  - warp
+- archive: ../acme-missiles-0.3.tar.gz
+  sha256: e563d8b524017a06b32768c4db8eff1f822f3fb22a90320b7e414402647b735b
+```
+
+With the `stack freeze` command, this would be replaced with:
+
+```yaml
+extra-deps:
+- size: 1540
+  url: https://hackage.haskell.org/package/acme-dont-1.1.tar.gz
+  cabal-file:
+    size: 602
+    sha256: 8264ad3e5113d3e0417b46e71d5a9c0914a1f03b5b81319cc329f1dc0f49b96c
+  name: acme-dont
+  version: '1.1'
+  sha256: c32231ff8548bccd4f3bafcc9b1eb84947a2e5e0897c50c048e0e7609fc443ce
+  pantry-tree:
+    size: 206
+    sha256: 79dbeddaf0fd507611687cefe9511c8fda489849fb0cac3894925716936290b2
+- size: 285152
+  subdir: wai
+  url: http://github.com/yesodweb/wai/archive/2f8a8e1b771829f4a8a77c0111352ce45a14c30f.zip
+  cabal-file:
+    size: 1717
+    sha256: 7b46e7a8b121d668351fa8a684810afadf58c39276125098485203ef274fd056
+  name: wai
+  version: 3.0.2.3
+  sha256: 3b6eb04f3763ca16432f3ab2135d239161fbe2c8811b8cd1778ffa67469289ba
+  pantry-tree:
+    size: 10296
+    sha256: ce431f1a22fcda89375ba5e35e53aee968eea23d1124fcba7cb9eae426daa2db
+- size: 285152
+  subdir: warp
+  url: http://github.com/yesodweb/wai/archive/2f8a8e1b771829f4a8a77c0111352ce45a14c30f.zip
+  cabal-file:
+    size: 6648
+    sha256: e3f01fd7417af923fd30962e9e6a4fe4de41ebc5e02af9819067fed79c9c6575
+  name: warp
+  version: 3.0.13.1
+  sha256: 3b6eb04f3763ca16432f3ab2135d239161fbe2c8811b8cd1778ffa67469289ba
+  pantry-tree:
+    size: 4292
+    sha256: d6b1def306a042b5fc500930302533a3ea828e916c99cbd82c0b7e2c4e3a8e09
+- size: 1442
+  filepath: acme-missiles-0.3.tar.gz
+  cabal-file:
+    size: 613
+    sha256: 2ba66a092a32593880a87fb00f3213762d7bca65a687d45965778deb8694c5d1
+  name: acme-missiles
+  version: '0.3'
+  sha256: e563d8b524017a06b32768c4db8eff1f822f3fb22a90320b7e414402647b735b
+  pantry-tree:
+    size: 226
+    sha256: 614bc0cca76937507ea0a5ccc17a504c997ce458d7f2f9e43b15a10c8eaeb033
+```
+
+## Snapshots
```

Comment:
Purescript's `psc-package` has a [verify command](https://psc-package.readthedocs.io/en/latest/maintenance.html) that ensures a snapshot (purescript: package set) builds correctly. Will this be in the curator tool?

Reply:
Yes, the goal of `curator` is to:

* Create a snapshot
* Test the snapshot
* Publish the snapshot